### PR TITLE
Turn to note midline only

### PR DESCRIPTION
**Pull request by codejvn**
results of the method i added:
true meaning it's a midline auto, false meaning it's not a midline auto

axis_n6: true
axis_n7: true
n4_return: false
n5_return: false
n6_return_under: false
n7_return_out: false
n7_return_short: false
n7_return_under: false
n7_return_under_n2: false
n7_return_under_short: false
n8_return: false
n8_return_short: false
n8_return_under: false
note1_n2: false
note1_n4: true
note1_n5: true
note2_n1: false
note2_n3: false
note2_n4: true
note2_n5: true
note2_n6: true
note2_n7: true
note3_n2: false
note3_n6: true
note3_n7: true
note3_n7_under: true
note3_n8: true
note5_under: true
redundant: false
redundant_1: false
short_droppedD: false
short_n7: true
short_n8: true
startA_n1: false
startA_n4: true
startA_note1: false
startB: false
startB_note1: false
startB_note2: false
startB_note3: false
startC: false
startC_forward: false
startC_n8: true
startC_note3: false
startD: false
startD_n7: true
startD_n8: true

### CODE REVIEW

File: src/main/java/frc/robot/autos/AutoRoutines.java [hunk 1]
```diff
@@ -385,4 +387,26 @@ private static final Optional<Rotation2d> turnToNoteOverride() {
...
+   * This works with our current naming convnetion (having the target note as the last number)
+   * 
+   * @param pathName - String name of a path
+   * @return true if it's a path that ends at the midline, otherwise false
+   */
+  private static boolean isMidlineAuto(String pathName){
+    if (!pathName.contains("return")) {
+      for (int i = pathName.length(); i > 1; i--) {
+          try {
+              if (Integer.parseInt(pathName.substring(i - 1, i)) >= 4) {
```
Comment (asimfazal) [inline src/main/java/frc/robot/autos/AutoRoutines.java:402]: It's a good idea but it's expensive to do. For cases non-midline names (notes 1 to 3), this will traverse the entire string (not the expensive part) but converting to int (parseInt) is expensive.

One way to optimize is to use Character.isDigit(), something like:

      for (int i = pathName.length(); i >1; i--) {
            char ch = input.charAt(i);
            if (Character.isDigit(ch) ) {
              int noteNumber = ch - '0';
               return noteNumber >= 4;
            }
        }
       return false;

Basically, it will stop at the first number it finds from the back of the string and returns true only if the number is >=4
